plugins/copyfilelocation.lua

```diff
@@ -1,15 +1,35 @@
 -- mod-version:3
 local core = require "core"
 local command = require "core.command"
+local treeview = require 'plugins.treeview'
+local menu = treeview.contextmenu
 
-command.add("core.docview", {
-  ["copy-file-location:copy-file-location"] = function(dv)
-    local doc = dv.doc
+command.add("core.docview!", {
+  ["doc:copy-file-location"] = function(view)
+    local doc = view.doc
     if not doc.abs_filename then
       core.error "Cannot copy location of unsaved doc"
       return
     end
-    core.log("Copying to clipboard \"%s\"", doc.abs_filename)
+    core.log("Copying \"%s\" to clipboard", doc.abs_filename)
     system.set_clipboard(doc.abs_filename)
   end
 })
+
+command.add(function()
+  return treeview.hovered_item ~= nil
+end, {
+  ["treeview:copy-file-location"] = function()
```

Comment:
Having the plugin name here too, might be a good idea.

Reply:
but doesn't that overwrite the previous command of the same name?

Reply:
You could call it `copy-file-location:copy-treeview-item-location` or something like that.